docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 			description: "Raised when the provided input cannot be parsed."
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
+		"array": {
+			description: "A list of items."
+			use: ["parameter", "return"]
+		}
+		"boolean": {
+			description: "`true` or `false`."
+			use: ["parameter", "return"]
+		}
+		"float": {
+			description: "A 64-bit floating-point number."
+			use: ["parameter", "return"]
+		}
+		"map": {
+			description: """
+				A key-value map in which keys are strings and values can be of any TRL type,
+				including maps.
+				"""
+			use: ["parameter", "return"]
+		}
+		"integer": {
+			description: "A 64-bit integer."
+			use: ["parameter", "return"]
+		}
+		"null": {
+			description: "No value."
+			use: ["return"]
+		}
+		"path": {
+			description: "An event field."
+			use: ["parameter"]
+		}
+		"regex": {
+			description: "A regular expression."
+			use: ["parameter"]
+		}
```

Comment:
Would love an example here, showing how to define regexes.

Reply:
```rust
$foo = /bar/
call_me($foo)
me_too(/baz/i)

.foo = /bar/ // compile-time error: cannot assign regex to object
```